src/components/BccInput/BccInput.vue

```diff
@@ -9,19 +9,28 @@ type Props = {
   state?: "default" | "error" | "success";
   disabled?: boolean;
   label?: string;
+  optionalLabel?: string;
+  required?: boolean;
 };
 
 withDefaults(defineProps<Props>(), {
   state: "default",
   disabled: false,
+  required: false,
 });
 </script>
 
 <template>
   <span class="inline-flex flex-col space-y-2">
-    <label class="bcc-input-label" v-if="label">{{ label }}</label>
+    <div>
+      <label class="bcc-input-label float-left" v-if="label">{{ label }}</label>
+      <span class="bcc-input-optional float-right" v-if="required === false">{{
```

Comment:
Instead of floats putting this in a flexbox container with `justify-items: in-between` would be a more modern approach.

```suggestion
    <div class="flex justify-between">
      <label class="bcc-input-label" v-if="label">{{ label }}</label>
      <span class="bcc-input-optional" v-if="required === false">{{
```

But maybe that `div` also needs a class.